src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```diff
@@ -40,6 +40,7 @@ public function isGranted(mixed $role, mixed $object = null, ?string $field = nu
             return false;
         }
 
+        // BC layer for Symfony AclBundle
         if (null !== $field) {
             $object = new FieldVote($object, $field);
```

Comment:
what about throwing a LogicException when the acl-bundle is not installed (as you suggested, isn't it?)

Reply:
I saw it as two distinct questions. This one really is only about having the same behaviour in both methods.

As `FieldVote` is in `symfony/security-acl` ..  detecting the Bundle will require injecting this information (and i hoped to not _add_ anything for something we "remove")  Or maybe we can consider the `FieldVote` class existence is enough here?

Something like this?

```php
if (3 === func_num_args() && !class_exists(FieldVote::class)) {
    throw new \LogicException('Passing a third argument to "is_granted" function requires the FieldVote class and the ACLBundle.');
}
```

Genuine question: is there any way here to word this without pushing for the bundle ?

Reply:
Why are we so obsessed with AclBundle? **If** this is useful for AclBundle, I bet that some other public or private code also use it, right? If yes, then, there is nothing to deprecate nor remove. It not, let's deprecate. Sorry if I'm missing something here (as I must admit that I haven't checked the code).

Reply:
This PR was only about aligning the newly `is_granted_for_user`  handling of this 3rd arg with the existing one in `is_granted`

You guys told me compatibility with AclBundle is required, i've nothing to add on this.

Just working on the DX here.

As i tried to explain on the original issue: 
* the `is_granted` function accepts 3 arguments. 
* the third is neither tested nor documented (but listed from many places [here](https://symfony.com/doc/current/reference/twig_reference.html#is-granted) )
* when provided, this 3rd arg is passed to the `FieldVote` __constructor (from symfony/security-acl)
* the existence of this class is not checked

My only agenda here is to clarify the situation. And prevent frustrations.

Reply:
`if (!class_exists(FieldVote::class)) {` inside the existinf "if" looks enough to me, then I'd suggest:

> Passing a $field to the "is_granted()" function requires symfony/acl. Try running "composer require symfony/acl-bundle" if you need field-level access control.